example/main.ts

```diff
@@ -237,5 +340,73 @@ function changeSlotTexture(selectItem, textures, spineAnimation) {
   const slotKey = selectItem.split("_")[0];
   const slotName = slotNameMap[slotKey];
   const index = selectItem.split("_")[1];
+  console.log(slotKey, slotName, index);
   spineAnimation.hackSeparateSlotTexture(slotName, textures[index]);
 }
+
+function removeController() {
+  if (animationController) {
+    animationController.remove();
+    animationController = null;
+  }
+  if (skinController) {
+    skinController.remove();
+    skinController = null;
+  }
+  if (slotHackController1) {
+    slotHackController1.remove();
+    slotHackController1 = null;
+  }
+  if (slotHackController2) {
+    slotHackController2.remove();
+    slotHackController2 = null;
+  }
+  if (slotHackController3) {
+    slotHackController3.remove();
+    slotHackController3 = null;
+  }
+  if (attachmentController) {
+    attachmentController.remove();
+    attachmentController = null;
+  }
+}
+
+window.onload = function() {
+  const fileInput = document.getElementById('fileInput') as HTMLInputElement;
+  const linkContainer = document.getElementById('linkContainer');
+  fileInput.addEventListener('change', function(event) {
+    const files = fileInput.files;
+    if (files) {
+      // Clear previous links
+      if (linkContainer) {
+        linkContainer.innerHTML = '';
+      }
+
+      // Create and display a temporary link for each file
+      Array.from(files).forEach((file, index) => {
+        const tempLink = URL.createObjectURL(file);
+        const ext = getFileExtension(file);
+        blobResource.urls.push(tempLink);
+        blobResource.params.fileExtensions.push(ext);
+        console.log(blobResource);
+      });
+    }
+  });
+};
+
+function getFileExtension(file: File): string {
+  const fileName = file.name;
+  const lastDotIndex = fileName.lastIndexOf('.');
+  if (lastDotIndex === -1 || lastDotIndex === 0) {
+    return '';
+  }
+  return fileName.substring(lastDotIndex + 1);
+}
+
+function delay(time: number) {
+  return new Promise((res) => {
+    setTimeout(() => {
+      res(true);
+    }, time);
+  });
+}
```

Comment:
It is recommended to use e2e test, e2e comes with case

Reply:
done